### PR TITLE
Snippet for sklearn

### PR DESCRIPTION
**Pull request by merveenoyan**
Taking from https://github.com/huggingface/hub-docs/pull/314 thought it would be better to have a separate branch instead of fork given I also change moon-landing code

### CODE REVIEW

**Comment by merveenoyan** [comment]
I somehow can't pass config to ModelInfo properly in Models.ts.

```
const x: SetOptional<ModelInfoWithSiblings, "siblings"> = new ModelInfo({
			_id:                 repo._id,
			id:                  repo.name,
			author:              repo.author,
			sha:                 commit.id,
			lastModified:        commit.author.date,
			lastAuthor:          { email: commit.author.email },
			private:             repo.config.private,
			gated:               repo.config.gated,
			discussionsDisabled: repo.config.discussionsDisabled,
			gitalyUid:           repo.gitalyUid,
			files,
			config,
		});
console.log({ x });
console.log({ config });
```
config is still { config: { filename: 'model.pkl' } } but in ModelInfo it's {}. Do you have any idea?

**Comment by osanseviero** [comment]
I'm not entirely sure to what does the code you're sharing correspond to. My workflow is to create a model repo with the same `README.md` and `config.json` as you would have in production, and within the `sklearn` method you can inspect the `model` by printing it. 

You're doing 
```
const skopsmodelFile = model.config.filename
```

But the model file should be within `model.sklearn`, as it's done for `adapter_transformers` and `speechbrain`. E.g. compare your internal code against
* https://github.com/huggingface/moon-landing/pull/4004/files#diff-70e0418cb1c6bf88268219ab03cc6b00d76364960e8ab99671c71bfeb8f78ff4R347-R352
* https://github.com/huggingface/moon-landing/pull/4004/files#diff-70e0418cb1c6bf88268219ab03cc6b00d76364960e8ab99671c71bfeb8f78ff4R359-R361

For debugging you can also add a `console.log(config)` in the internal PR to understand what's going on, but there you're fully overriding the config.

File: js/src/lib/interfaces/Libraries.ts [hunk 1]
```diff
@@ -218,13 +218,22 @@ const timm = (model: ModelData) =>
...
+download("${model.id}", "path_to_folder")
+model = load(${skopsmodelFile})`;
+	} else {
+		return `from huggingface_hub import hf_hub_download
 import joblib
-
 model = joblib.load(
 	hf_hub_download("${model.id}", "sklearn_model.joblib")
-)`;
+			)`;
```
Comment (osanseviero) [inline js/src/lib/interfaces/Libraries.ts:233]: The indentation for this looks off in the screenshot you shared

**Review by osanseviero — CHANGES_REQUESTED** [review]
This looks good! Thanks for working on it! I left some minor suggestions and improvement requests

**Comment by osanseviero** [comment]
Let me know if it's ready for another review (and remember to mark as solved if things are resolved)

File: js/src/lib/interfaces/Libraries.ts [hunk 2]
```diff
@@ -218,13 +218,24 @@ const timm = (model: ModelData) =>
...
-const sklearn = (model: ModelData) =>
-	`from huggingface_hub import hf_hub_download
+const sklearn = (model: ModelData) => {
+	if (model.tags?.includes("skops")) {
+		const skopsmodelFile = model.config?.sklearn?.filename;
+		return `from skops.hub_utils import download
+from skops.io import load
+
+download("${model.id}", "path_to_folder")
+model = load(${skopsmodelFile})`;
```
Comment (osanseviero) [inline js/src/lib/interfaces/Libraries.ts:228]: Does copy pasting this snippet work? Should `load` also pass the folder where it was saved?

**Comment by merveenoyan** [comment]
Ekran Resmi 2022-10-04 14 38 49
@osanseviero WDYT

File: js/src/lib/interfaces/Libraries.ts [hunk 3]
```diff
@@ -218,13 +218,26 @@ const timm = (model: ModelData) =>
...
-	`from huggingface_hub import hf_hub_download
+const sklearn = (model: ModelData) => {
+	if (model.tags?.includes("skops")) {
+		const skopsmodelFile = model.config?.sklearn?.filename;
+		return `from skops.hub_utils import download
+from skops.io import load
+
+download("${model.id}", "path_to_folder")
+# make sure model file is in skops format
+# if model is a pickle file, load with pickle at your own risk
```
Comment (osanseviero) [inline js/src/lib/interfaces/Libraries.ts:229]:
```suggestion
# if model is a pickle file, make sure it's from a source you trust
```

**Review by osanseviero — APPROVED** [review]
Looks good! I just left a suggestion on the comment :fire: thanks!

**Pull request merged**